### DISCUSSION TITLE
Multiple Devices Same "U"

### DISCUSSION THREAD

**Discussion by cristgal** [netbox-community/netbox#8021] [General]
I have the same issue as "Multiple devices in rack unit #1096" but not sure how to create a "shelf".

Some of my chasses are 2U / four servers:
```
U2    [srv1] [srv2]
U1    [srv3] [srv4]
```

but just FYI, in the past I've used vertical 4U servers:
```
U4  [S] [S] [S] [S] [S] [S] [S] [S] 
U3  [R] [R] [R] [R] [R] [R] [R] [R] 
U2  [V] [V] [V] [V] [V] [V] [V] [V] 
U1  [1] [2] [3] [4] [5] [6] [7] [8]
```

Thoughts on how to describe it into the rack?
thanks!

**Comment by candlerb** [comment]
That's exactly what "parent and child" devices are intended to model.

* Create a [device type](https://netbox.readthedocs.io/en/stable/core-functionality/device-types/) for the outer chassis, size 2U or 4U, type "parent", with 4 or 8 device bays
* Create an inner device type for the server blade, type "child"

You can then install the parent device in the rack, and the child devices in the parent device bays.

This works pretty, well although there are a few limitations:
* The rack elevations don't display the component servers or blades
* Sometimes there isn't an actual physical parent device, but you have to make an imaginary one anyway.  (Example: two half-width switches which bolt together side-by-side to fit in the same 1U rack unit).
* It gets a bit ugly if you choose to model "devices on a shelf" this way.  Personally I don't do that: I assign those devices to the rack without assigning them to any rack position, which displays them as "unracked", and put a note in the comments.

Also, be aware that the device bay model is *not* for use for plug-in modules within the same device, like line cards in a chassis router.  It's only for independent, separately managed devices, which just happen to live in the same housing.  Your example of separate servers within a housing is fine.  If you plug in extra modules to a device, like line cards or PCI cards or whatever, these should be modelled as "Inventory Items".
